gems/pending/metadata/linux/LinuxUtils.rb

```diff
@@ -141,6 +141,44 @@ def self.parse_systemctl_list(lines)
       end.compact!
     end
 
+    def self.collect_interface(interfaces, interface)
+      mac_addr = interface[:mac_address]
+      return if mac_addr.blank?
+
+      existing_interface = interfaces[mac_addr]
+      if existing_interface.blank?
+        interfaces[mac_addr] = interface
+      else
+        interface[:name] += "," + existing_interface[:name]
+        existing_interface.merge!(interface)
+      end
+    end
+
+    def self.parse_network_interface_list(lines)
```

Comment:
With this kind of parsing, including an example of the lines to be parsed can go a long way.  For instance:

```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: em1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether ff:ff:ff:ff:ff:ff brd ff:ff:ff:ff:ff:ff
    inet 1.1.1.1/23 brd 1.1.1.0 scope global dynamic em1
       valid_lft 62383sec preferred_lft 62383sec
    inet6 1:1:1:1:1:1:1:1/64 scope global noprefixroute dynamic 
       valid_lft 2591984sec preferred_lft 604784sec
```